### PR TITLE
Proper message for problem with addons

### PR DESCRIPTION
**Pull request by orihomie**
## Motivation/Description of the PR

Getting frustrated when I see some unreadable error.

- Logging what went wrong

## Type of change

- [ ] Breaking changes (its not)

### CODE REVIEW

File: lib/listener/helpers.js [hunk 1]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
...
+      return;
+    
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+      if (!helpers[key][hook]) {
+			  console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
+			  return;
+		  }
+
+		  helpers[key][hook](param);
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579&bid=16313621)

File: lib/listener/helpers.js [hunk 2]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
   const helpers = container.helpers();
 
   const runHelpersHook = (hook, param) => {
-    if (store.dryRun) return;
+    if (store.dryRun) 
+      return;
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Expected no linebreak before this statement.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 3]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
...
+    
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+      if (!helpers[key][hook]) {
+			  console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
+			  return;
+		  }
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 4]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
   const helpers = container.helpers();
 
   const runHelpersHook = (hook, param) => {
-    if (store.dryRun) return;
+    if (store.dryRun)
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Expected { after 'if' condition.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 5]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
...
   const runHelpersHook = (hook, param) => {
-    if (store.dryRun) return;
+    if (store.dryRun) 
+      return;
+    
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+      if (!helpers[key][hook]) {
+			  console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
+			  return;
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 6]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
...
 
   const runHelpersHook = (hook, param) => {
-    if (store.dryRun) return;
+    if (store.dryRun) 
+      return;
+    
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+      if (!helpers[key][hook]) {
+			  console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 7]
```diff
@@ -11,9 +11,16 @@ module.exports = function () {
   const helpers = container.helpers();
 
   const runHelpersHook = (hook, param) => {
-    if (store.dryRun) return;
+    if (store.dryRun) 
+      return;
+
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 3 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 1 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 4 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 6 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579&bid=16313621)

File: lib/listener/helpers.js [hunk 8]
```diff
@@ -13,7 +13,11 @@ module.exports = function () {
   const runHelpersHook = (hook, param) => {
     if (store.dryRun) return;
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+    	if (!helpers[key][hook]) {
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 9]
```diff
@@ -13,7 +13,11 @@ module.exports = function () {
...
+    		console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
+			  return;
+		  }
+		  helpers[key][hook](param);
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 10]
```diff
@@ -13,7 +13,11 @@ module.exports = function () {
   const runHelpersHook = (hook, param) => {
     if (store.dryRun) return;
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+    	if (!helpers[key][hook]) {
+    		console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 11]
```diff
@@ -13,7 +13,11 @@ module.exports = function () {
...
+			  return;
+		  }
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 12]
```diff
@@ -13,7 +13,11 @@ module.exports = function () {
   const runHelpersHook = (hook, param) => {
     if (store.dryRun) return;
     Object.keys(helpers).forEach((key) => {
-      helpers[key][hook](param);
+    	if (!helpers[key][hook]) {
+    		console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
+			  return;
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 8 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 12 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 9 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

**Comment by orihomie** [comment]
Can someone please help me to for that in a proper way?

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579&bid=16313621)

File: lib/listener/helpers.js [hunk 13]
```diff
@@ -13,6 +13,10 @@ module.exports = function () {
   const runHelpersHook = (hook, param) => {
     if (store.dryRun) return;
     Object.keys(helpers).forEach((key) => {
+      if (!helpers[key][hook]) {
+	    console.log(`${key}'s hook value is: ${helpers[key][hook]}`);
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js [hunk 14]
```diff
@@ -13,6 +13,10 @@ module.exports = function () {
...
+	    return;
```
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 13 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Unexpected tab character.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

File: lib/listener/helpers.js (hunk 14 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4932579)

**Comment by PeterNgTr** [comment]
> Can someone please help me to fork that in a proper way?

@orihomie clicking fork button didn't work for you?

**Comment by orihomie** [comment]
> > Can someone please help me to fork that in a proper way?
> 
> @orihomie clicking fork button didn't work for you?

Sry I was talking about that Hound violations

**Comment by PeterNgTr** [comment]
> > > Can someone please help me to fork that in a proper way?
> > 
> > 
> > @orihomie clicking fork button didn't work for you?
> 
> Sry I was talking about that Hound violations

better to run on your local with `npm run lint-fix` that will fix all possible lint issues.

File: lib/listener/helpers.js (hunk 13 above)
Comment (DavertMik) [inline lib/listener/helpers.js]: please use `output.log` from `require('../output')`;
you will be able to see that message when executing in `--verbose` mode.

Also you can use `debug` calls from `debug` module
But not console.log!

File: lib/listener/helpers.js [hunk 15]
```diff
@@ -13,6 +13,10 @@ module.exports = function () {
   const runHelpersHook = (hook, param) => {
     if (store.dryRun) return;
     Object.keys(helpers).forEach((key) => {
+      if (!helpers[key][hook]) {
+        error(`${key}'s hook value is: ${helpers[key][hook]}`);
```
Comment (DavertMik) [inline lib/listener/helpers.js]:
```suggestion
```
I don't see any point in this message. You checked for this value and it is null or undefined
better to remove this check so any class could be a helper

**Pull request merged**